tinygrad/lazy.py

```diff
@@ -372,7 +372,7 @@ def _realize_rand(buffer: LazyBuffer) -> None:
   buffer.realized = Device[buffer.device].buffer.fromCPU(rng.random(size=buffer.shape, dtype=np.float32).astype(dtype=buffer.dtype.np, copy=False), **buffer._device_extra_args()) # type: ignore
 
 def _realize_const(buffer: LazyBuffer) -> None:
-  if isinstance(Device[buffer.device], Compiled) and buffer.device not in ["LLVM"]:  # consts are broken in LLVM in NaN/inf
+  if isinstance(Device[buffer.device], Compiled) and not (buffer.device == "LLVM" and not math.isfinite(buffer.op.arg)): # consts are broken in LLVM in NaN/inf
```

Comment:
Only inf and nan fail for LLVM. But to get the optimizations for LLVM we want the constant.